### PR TITLE
Add udocker

### PR DESCRIPTION
**Pull request by sebastian-luna-valero**
Checklist
- [X] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [X] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/main/recipes/example/meta.yaml#L57-L66) for an example).
- [X] Source is from official source.
- [X] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [X] If static libraries are linked in, the license of the static library is packaged.
- [X] Build number is 0.
- [X] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [X] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [X] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by sebastian-luna-valero** [comment]
Hi @jorge-lip and @mariojmdavid

Could you please let me know whether you are happy with this recipe to add `udocker` to `conda-forge`? Would you like to be listed as maintainer of this package?

Best regards,
Sebastian

**Comment by mariojmdavid** [comment]
hi Sebastian
latest version is 1.3.1 and is in pypi
Jorge of course the main maintainer, and I contributer, maybe can also be stated as maintainer
will check the recipe

**Comment by mariojmdavid** [comment]
version 1.3.1 supports python >= 3.6
and will have to familiarize with these recipes I think

**Comment by sebastian-luna-valero** [comment]
Hi Mario,

The recipe is in the `recipes/udocker/meta.yaml` file:
* ` {% set version = "1.3.1" %}` is where I specify which version of `udocker` to download
* `url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz` points to the source code on PyPI
* `skip: True  # [py<36]` ensures this package is built for Python 3.6, 3.7, 3.8 and 3.9

Please note that I am patching `setup.py` with `recipes/udocker/setup.py.patch` to remove a couple of things:
* `use_2to3=True,`: it gave me errors and it looks like it's passing tests without it, is it safe to remove it?
* I think `pytest-runner` does not need to be part of `SETUP_REQUIREMENTS = ['pytest-runner', ]`?

Many thanks,
Sebastian

File: recipes/udocker/meta.yaml
Comment (dopplershift) [inline recipes/udocker/meta.yaml:16]: This looks like it's pure Python and could be noarch Python?
```suggestion
  noarch: python
```

File: recipes/udocker/meta.yaml
Comment (dopplershift) [inline recipes/udocker/meta.yaml:25]: If noarch:
```suggestion
    - python >=3.6
#    - pytest-runner # ERROR: No matching distribution found for pytest-runner
    - pip
  run:
    - python >=3.6
```

**Comment by sebastian-luna-valero** [comment]
Many thanks @dopplershift. I have just pushed the requested changes.

I did have a look at https://conda-forge.org/docs/maintainer/knowledge_base.html#noarch-python but I wasn't completely sure to meet all requirements.

**Comment by dopplershift** [comment]
noarch looks fine, but you have some test failures to deal with.

**Comment by sebastian-luna-valero** [comment]
True, but tests are passing for Linux which I think is the only target of this package.

Let's wait for the developers' feedback about fixing or disabling the tests for OSX and Windows.

**Comment by sebastian-luna-valero** [comment]
Hi @jorge-lip and @mariojmdavid

Could you please confirm whether you would like to have a conda package for `udocker` (i.e. keep working on this)? or should we close this PR?

Best regards,
Sebastian

**Comment by mariojmdavid** [comment]
will work on this, apologies for the delay

**Comment by sebastian-luna-valero** [comment]
@mariojmdavid apologies, I don't know why I missed this notification

I am updating this recipe now with `udocker-1.3.3` recently released.

Also I found a similar PR to add `udocker` to `conda-forge`: https://github.com/conda-forge/staged-recipes/pull/18751

**Comment by sebastian-luna-valero** [comment]
@mariojmdavid could you please have a look at the errors in the tests [here](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=558543&view=logs&j=6f142865-96c3-535c-b7ea-873d86b887bd&t=22b0682d-ab9e-55d7-9c79-49f3c3ba4823&l=1355).

**Comment by mariojmdavid** [comment]
hi Sebastian, I have fixed the unit test, and commit to master in udocker,
how do we trigger the tests here again?

**Comment by sebastian-luna-valero** [comment]
Hi Mario,

Tests are passing from the main branch. Could you please make a new release? This would be required for the conda package.

Best regards,
Sebastian

**Comment by mariojmdavid** [comment]
new version 1.3.4 in pypi
and github master

**Comment by sebastian-luna-valero** [comment]
Thanks @mariojmdavid!

We just need a review from the `conda-forge` team to get this merged.

**Comment by sebastian-luna-valero** [comment]
@conda-forge/staged-recipes  please review

**Review by carterbox — APPROVED** [review]
Although this is a pure python package which could be noarch. The implementation assumes that the host system is linux and will not work on non-linux. I think it's less confusing for users if the package is not installable on non-linux platforms. If noarch: python linux-only were a thing, I would choose that.

**Comment by BastianZim** [comment]
> If noarch: python linux-only were a thing, I would choose that.

You can add the `__linux` virtual package?

Ref: https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-virtual.html

**Comment by carterbox** [comment]
Ahh! I forgot those existed! I will open a PR. Good call, @BastianZim!

**Pull request merged**
